### ISSUE TITLE
Add support for soundex algorithm

### ISSUE THREAD

**Issue by rayjlinden** [moov-io/ofac#150]
The Treasury's UI also calculates the score used with the phonetic based algorithm Soundex.
https://www.treasury.gov/resource-center/sanctions/SDN-List/Pages/fuzzy_logic.aspx

This algo should be supported as well and incorporated into the overall search score.

**Comment by baldurh** [comment]
Are there still plans to add soundex / phonetics support to this project?

**Comment by adamdecaf** [comment]
Yes. I haven't looked for a good implementation to use lately. Do you know of one? It's not too bad to implement if needed.

**Comment by baldurh** [comment]
I haven’t looked into that tbh. I was just wondering since we’re comparing this to our current solution 🙂

**Comment by adamdecaf** [comment]
More reading:
- https://aclanthology.org/W04-2905.pdf
- https://arxiv.org/pdf/1405.6293.pdf
- https://airccse.org/journal/ijcsea/papers/4314ijcsea03.pdf
- https://www.basistech.com/whitepapers/the-name-matching-you-need-EN.pdf
- https://www.iijg.org/wp-content/uploads/2014/02/Alexander_Beider.pdf

**Comment by MorganaFuture** [comment]
@adamdecaf  Is this task still relevant?

**Comment by adamdecaf** [comment]
@MorganaFuture yea, but less so lately I assume. We can support an `?algorithm=soundex` parameter and env variable. That way callers can either change the default or on a per-request basis.

**Comment by samofolabi** [comment]
Hey man @adamdecaf I am interested in this.
